network/p2p/unicast/rate_limiters.go

```diff
@@ -0,0 +1,102 @@
+package unicast
+
+import (
+	"github.com/libp2p/go-libp2p/core/peer"
+
+	"github.com/onflow/flow-go/network/channels"
+	"github.com/onflow/flow-go/network/message"
+)
+
+var (
+	MessageCount RateLimitReason = "messagecount"
+	Bandwidth    RateLimitReason = "bandwidth"
+)
+
+type RateLimitReason string
+
+func (r RateLimitReason) String() string {
+	return string(r)
+}
+
+type OnRateLimitedPeerFunc func(pid peer.ID, role, msgType string, topic channels.Topic, reason RateLimitReason) // the callback called each time a peer is rate limited
+
+// RateLimiters used to manage stream and bandwidth rate limiters
+type RateLimiters struct {
+	MessageRateLimiter   RateLimiter
+	BandWidthRateLimiter RateLimiter
+	OnRateLimitedPeer    OnRateLimitedPeerFunc // the callback called each time a peer is rate limited
+	dryRun               bool
```

Comment:
Two comments about this:
1. The term "dry run" is a general name and a bit unclear, it would be great to rename it to something more self-descriptive and support it with documentation. 
2. This variable seems to be `true` by default in production and `false` in testing. Hence, it makes sense to let it be initialized to the default value and override it with an option method whenever needed (e.g., during tests).

Reply:
https://github.com/onflow/flow-go/pull/2840/commits/ea2cc01064ef77b481051b4eac33f89869f47e97